[quant][pt2e][improvement] pt2e quant flow preserve metadata and graph module type

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99549

Summary:
att, this is to allow the quant flow to work with graph module produced by
exir capture (not yet oss'ed), it stores some type information for input and output
in model.meta, so we need to preserve these in the meta field

Test Plan:
will be tested internally to make sure users of pt2e quant flow don't need to
explicitly preserve metadata when they use this flow

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][pt2e][improvement] pt2e quant flow preserve metadata and graph module type

Summary:
att, this is to allow the quant flow to work with graph module produced by
exir capture (not yet oss'ed), it stores some type information for input and output
in model.meta, so we need to preserve these in the meta field

Test Plan:
will be tested internally to make sure users of pt2e quant flow don't need to
explicitly preserve metadata when they use this flow

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel reviewed (approved):
Looks good, but I think we gonna have to go a step further and figure out how to preserve this info for observer nodes and quant/dequant nodes. Shouldnt be very hard, but I want to think through it so that it is meaningful for backend lowering.

jerryzh168 committed:
Update on "[quant][pt2e][improvement] pt2e quant flow preserve metadata and graph module type"


Summary:
att, this is to allow the quant flow to work with graph module produced by
exir capture (not yet oss'ed), it stores some type information for input and output
in model.meta, so we need to preserve these in the meta field

Test Plan:
will be tested internally to make sure users of pt2e quant flow don't need to
explicitly preserve metadata when they use this flow

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
this doesn't work, we'll work with torch.export together to solve this problem in the future
